src/components/Tool/Appeal/Modals/PledgeModal/PledgeModal.tsx

```diff
@@ -114,6 +118,22 @@ export const PledgeModal: React.FC<PledgeModalProps> = ({
           },
         },
         refetchQueries: ['Contacts', 'Appeal'],
+        update: (_, data) => {
```

Comment:
Semantically, `update` functions are for updating the cache. Can you move this logic to `onCompleted`?